google/oauth2/_client.py

```diff
@@ -59,7 +66,41 @@ def _handle_error_response(response_data):
     except (KeyError, ValueError):
         error_details = json.dumps(response_data)
 
-    raise exceptions.RefreshError(error_details, response_data)
+    raise exceptions.RefreshError(
+        error_details, response_data, retryable=retryable_error
+    )
+
+
+def _can_retry(status_code, response_data):
+    """Checks if a request can be retried by inspecting the status code
+    and response body of the request.
+
+    Args:
+        status_code (int): The response status code.
+        response_data (Mapping | str): The decoded response data.
+
+    Returns:
+      bool: True if the response is retryable. False otherwise.
+    """
+    try:
+        # For a failed response, response_body could be a string
+        error_desc = response_data.get("error_description") or ""
+        error_code = response_data.get("error") or ""
+
+        # Per Oauth 2.0 RFC https://www.rfc-editor.org/rfc/rfc6749.html#section-4.1.2.1
+        # This is needed because a redirect will not return a 500 status code.
+        retryable_error_descriptions = {"internal_failure", "server_error", "temporarily_unavailable"}
```

Comment:
Why we still have it? we have a status code check below, those are more reliable than error descriptions

Reply:
My understanding from the spec is that these errors _may_ service as a non-500 status code in case the request was redirected.

Reply:
+ it seems to be related to 3LO that is not supported

Reply:
i would check how this was added, as original implementation or related to a fix

Reply:
@TimurSadykov IIUC this is the original implementation https://github.com/googleapis/google-auth-library-python/pull/368.

Bug report https://bugs.chromium.org/p/chromium/issues/detail?id=1001206